petclock.asm

```diff
@@ -1012,6 +1153,191 @@ GetDeviceStatus:
 .endif          ; PETSDPLUS
 
 
+.if C64         ; Only C64 has 6526 CIAs
+
+;----------------------------------------------------------------------------
+; InitCIAClock - Initialize CIA clock to correct external frequency (50/60Hz)
+;----------------------------------------------------------------------------
+
+InitCIAClock:
+                lda CIA2_CRB
+                and #$7f                ; Set CIA2 TOD clock, not alarm
+                sta CIA2_CRB
+
+                sei
+                lda #$00
+                sta CIA2_TENTHS         ; Start CIA2 TOD clock
+@tickloop:      cmp CIA2_TENTHS         ; Wait until tenths value changes

```

Comment:
I'm sure there's a reason, but.... why?

Reply:
This is to make sure the rest of the routine's code runs for exactly a tenth of a second, as measured by the TOD clock when set to an external frequency of 60 Hz. 
What this routine does as a whole is explained here: https://codebase64.org/doku.php?id=base:efficient_tod_initialisation